triedb/pathdb/history_index_block.go

```diff
@@ -0,0 +1,403 @@
+// Copyright 2025 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/
+
+package pathdb
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"sort"
+)
+
+const (
+	indexBlockDescSize   = 24    // The size of index block descriptor
+	indexBlockEntriesCap = 4096  // The maximum number of entries can be grouped in a block
+	indexBlockRestartLen = 256   // The restart interval length of index block
+	historyIndexBatch    = 65536 // The number of state histories for constructing or deleting indexes together
+)
+
+// indexBlockDesc represents a descriptor for an index block, which contains a
+// list of state mutation records associated with a specific state (either an
+// account or a storage slot).
+type indexBlockDesc struct {
+	min     uint64 // The minimum state ID retained within the block
+	max     uint64 // The maximum state ID retained within the block
```

Comment:
probably uint32 is sufficient?

Reply:
I think uint32 is cutting it pretty close
If we assume a block every second, we would reach ~140years with this. There are forks of geth that do sub second blocks iirc.

Reply:
Right, 0.25s means 34 years. I don't .. know ...